source/frontend/src/components/common/ApiVersionInfo.tsx

```diff
@@ -27,11 +32,78 @@ export const ApiVersionInfo = () => {
     };
   }, [getVersion]);
 
+  const findDBVersion = (frontEndVersion: string): string => {
+    // remove suffix
+    const frontEndVersionClean = frontEndVersion.substring(0, frontEndVersion.lastIndexOf('.'));
+    const start = frontEndVersion.lastIndexOf('-') + 1;
+    const end = frontEndVersion.lastIndexOf('.');
+
+    // Get DB Version.
+    const dbVersion = frontEndVersionClean.substring(start, end);
+    // Convert to 2 decimal version so it matches DB record.
+    const numericVersion = parseFloat(dbVersion).toFixed(2);
+
+    return numericVersion.toString();
+  };
+
+  const frontEndVersion = import.meta.env.VITE_PACKAGE_VERSION;
+  const frontEndDBVersion = findDBVersion(frontEndVersion);
+
+  const apiVersionMissmatch = version?.informationalVersion !== frontEndVersion;
+  const dbVersionMissmatch = version?.dbVersion !== frontEndDBVersion;
+
+  const versionMissmatchMsg = (apiMissmatch: boolean, dbMissmatch: boolean): string => {
+    let msg = '';
+    if (apiMissmatch || dbMissmatch) {
+      msg = msg.concat(
+        `Warning: There is a version missmatch with the backend.
+         API: ${version?.informationalVersion}; DB: ${version?.dbVersion}`,
+      );
+    }
+
+    return msg;
+  };
+
   return (
-    <div className="version" data-testid="version">
-      {version?.informationalVersion ? `v${version.informationalVersion ?? ''}` : 'api unavailable'}
-    </div>
+    <StyledContainer>
+      <div className="version" data-testid="version-tag">
+        {`v${frontEndVersion ?? ''}`}
+      </div>
+
+      {(apiVersionMissmatch || dbVersionMissmatch) && (
+        <VersionMissmatchDiv data-testid="version-missmatch-warning">
+          <TooltipWrapper
+            tooltipId="warning"
+            tooltip={versionMissmatchMsg(apiVersionMissmatch, dbVersionMissmatch)}
+            className="warning"
+          >
+            <AiOutlineExclamationCircle size={20} />
+          </TooltipWrapper>
+        </VersionMissmatchDiv>
+      )}
+    </StyledContainer>
   );
 };
 
 export default ApiVersionInfo;
+
+const StyledContainer = styled.div`
+  display: flex;
+  gap: 10px;
+  flex-grow: 1;
+  flex-direction: row;
+  justify-content: flex-end;
+  align-items: center;
+`;
+
+export const VersionMissmatchDiv = styled(InlineFlexDiv)`
+  text-transform: uppercase;
```

Comment:
I don't think this style is needed. We don't want the warning message in all caps

Reply:
removed